point p
---
float64 dist
